src/Ampersand/Input/Archi/ArchiAnalyze.hs

```diff
@@ -0,0 +1,705 @@
+                      [(P_Population, Maybe P_Relation, [PClassify])]
+     keyArchi   :: a -> String                      -- get the key value (dirty identifier) of an a.
+
+{- I'm not quite sure what the purpose of having eqAsy was in the first place...
+   instance Flippable P_Relation where
+     flp decl
+      = decl{ dec_sign =      flp  (dec_sign decl)
+            , dec_prps = (map flp) (dec_prps decl)
+            , dec_popu = (map flp) (dec_popu decl)
+            }
+
+   instance Flippable P_Population where
+     flp pop
+      = case pop of
+         P_RelPopu{} -> pop { p_src = p_tgt pop
+                            , p_tgt = p_src pop
+                            , p_nmdr = flp (p_nmdr pop)
+                            , p_popps = (map flp) (p_popps pop)
+                            }
+
+   instance Flippable P_NamedRel where
+     flp rel = rel{p_mbSign = flp (p_mbSign rel)}
+
+   eqAsy :: [(P_Population,  Maybe P_Relation, [PClassify])] -> [(P_Population, Maybe P_Relation, [PClassify])]
+   eqAsy [] = []
+   eqAsy ((pPop, maybeDecl, pgens): rest)
+    = ( foldr1 squash
+        ( (pPop, maybeDecl, pgens) : [ (flp pPop', flp maybeDecl', pgens')
+                                     | (pPop'@P_RelPopu{}, maybeDecl', pgens')<-rest
+                                     , name pPop'==name pPop, p_src pPop'==p_tgt pPop, p_tgt pPop'==p_src pPop, flp maybeDecl'==maybeDecl
+                                     ] ) )
+      : eqAsy [ (pPop', maybeDecl', pgens')
+              | (pPop'@P_RelPopu{}, maybeDecl', pgens')<-rest, name pPop'/=name pPop||p_src pPop'/=p_tgt pPop||p_tgt pPop'/=p_src pPop||flp maybeDecl'/=maybeDecl ]
+      where
+       squash :: (P_Population, Maybe P_Relation, [PClassify]) ->
+                 (P_Population, Maybe P_Relation, [PClassify]) ->
+                 (P_Population, Maybe P_Relation, [PClassify])
+       (pPp@P_RelPopu{}, mDecl, pgns) `squash` (pPp'@P_RelPopu{}, mDecl', pgns')
+        = ( pPp{p_popps = p_popps pPp `union` p_popps pPp'}
+          , case (mDecl, mDecl') of
+                 (Just decl, Just _)     -> Just (decl {dec_nm = dec_nm decl++"!"})
+                 (Just decl, Nothing)    -> Just (decl {dec_nm = dec_nm decl++"!"})
+                 (Nothing,   Just decl') -> Just (decl'{dec_nm = dec_nm decl'++"!"})
+                 _                       -> Nothing
+          , pgns `union` pgns'
+          )
+       (pPp@P_CptPopu{}, mDecl, pgns) `squash` (pPp'@P_CptPopu{}, mDecl', pgns')
+        = ( pPp{p_popas = p_popas pPp `union` p_popas pPp'}
+          , case (mDecl, mDecl') of
+                 (Just decl, Just _)     -> Just (decl {dec_nm = dec_nm decl++"!"})
+                 (Just decl, Nothing)    -> Just (decl {dec_nm = dec_nm decl++"!"})
+                 (Nothing,   Just decl') -> Just (decl'{dec_nm = dec_nm decl'++"!"})
+                 _                       -> Nothing
+          , pgns `union` pgns'
+          )
+       _ `squash` _ = fatal "error in squash"
+-}
```

Comment:
Is this code still of any value? If not, please remove it.

Reply:
done...